File: 2244-minimum-rounds-to-complete-all-tasks/2244-minimum-rounds-to-complete-all-tasks.cpp
class Solution {
public:
    int minimumRounds(vector<int>& tasks) {
        unordered_map<int, int> freq;
        for(auto task: tasks){
            freq[task]++;
        }
        int rounds = 0;
        for(auto [task, count]: freq){
            if(count == 1){
                return -1;
            }
            if(count%3 ==0){
                rounds += count/3;
            }else{
                rounds += count/3 + 1;
            }
        }
        return rounds;
    }
};

File: 2244-minimum-rounds-to-complete-all-tasks/2244-minimum-rounds-to-complete-all-tasks.py
class Solution:
    def minimumRounds(self, tasks: List[int]) -> int:
        counter = Counter(tasks)
        rounds = 0
        for val, count in counter.items():
            # print(val, count)
            if count == 1:
                return -1
            
            if count%3 == 0:
                #Group all the task in triplets.
                rounds += count//3
            else:
                # If count % 3 = 1; (count / 3 - 1) groups of triplets and 2 pairs.
                # If count % 3 = 2; (count / 3) groups of triplets and 1 pair.
                rounds += count//3 + 1
        return rounds